tiatoolbox/models/engine/io_config.py

```diff
@@ -0,0 +1,407 @@
+        output_resolutions (list(dict)):
+            Resolution of each output head from model inference, must be
+            in the same order as target model.infer_batch().
+        patch_output_shape (:class:`numpy.ndarray`, list(int)):
+            Shape of the largest output in (height, width).
+        save_resolution (dict):
+            Resolution to save all output.
+        highest_input_resolution (dict):
+            Highest resolution to process the image based on input and
+            output resolutions. This helps to read the image at the optimal
+            resolution and improves performance.
+        margin (int):
+            Tile margin to accumulate the output.
+        tile_shape (tuple(int, int)):
+            Tile shape to process the WSI.
+
+    Examples:
+        >>> # Defining io for a network having 1 input and 1 output at the
+        >>> # same resolution
+        >>> ioconfig = IOSegmentorConfig(
+        ...     input_resolutions=[{"units": "baseline", "resolution": 1.0}],
+        ...     output_resolutions=[{"units": "baseline", "resolution": 1.0}],
+        ...     patch_input_shape=(2048, 2048),
+        ...     patch_output_shape=(1024, 1024),
+        ...     stride_shape=(512, 512),
+        ... )
+
+    Examples:
+        >>> # Defining io for a network having 3 input and 2 output
+        >>> # at the same resolution, the output is then merged at a
+        >>> # different resolution.
+        >>> ioconfig = IOSegmentorConfig(
+        ...     input_resolutions=[
+        ...         {"units": "mpp", "resolution": 0.25},
+        ...         {"units": "mpp", "resolution": 0.50},
+        ...         {"units": "mpp", "resolution": 0.75},
+        ...     ],
+        ...     output_resolutions=[
+        ...         {"units": "mpp", "resolution": 0.25},
+        ...         {"units": "mpp", "resolution": 0.50},
+        ...     ],
+        ...     patch_input_shape=(2048, 2048),
+        ...     patch_output_shape=(1024, 1024),
+        ...     stride_shape=(512, 512),
+        ...     save_resolution={"units": "mpp", "resolution": 4.0},
+        ... )
+
+    """
+
+    margin: int = None
+    tile_shape: Tuple[int, int] = None
+
+    def to_baseline(self):
+        """Returns a new config object converted to baseline form.
+
+        This will return a new :class:`IOSegmentorConfig` where
+        resolutions have been converted to baseline format with the
+        highest possible resolution found in both input and output as
+        reference.
+
+        """
+        new_config = super().to_baseline()
+
+        return IOInstanceSegmentorConfig(
+            input_resolutions=new_config.input_resolutions,
+            output_resolutions=new_config.output_resolutions,
+            patch_input_shape=self.patch_input_shape,
+            patch_output_shape=self.patch_output_shape,
+            stride_shape=self.stride_shape,
+            save_resolution=new_config.save_resolution,
```

Comment:
```suggestion
            **new_config,
            patch_input_shape=self.patch_input_shape,
            patch_output_shape=self.patch_output_shape,
            stride_shape=self.stride_shape,
```

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/618/commits/caab9a924a0a2bd114aac9040ffb1768282ebc0a